language/evm/move-to-yul/tests/U256Arith.exp

```diff
@@ -0,0 +1,332 @@
+                // $t3 := 1701411834604692317316873037158841057281
+                /// @src 2:494:515
+                $t3 := 1701411834604692317316873037158841057281
+                // $t4 := -($t2, $t3)
+                /// @src 2:525:540
+                $t4 := $Sub($t2, $t3)
+                // return $t4
+                /// @src 2:525:540
+                $result := $t4
+            }
+
+            function A2_U256Arith_cast(x, y) -> $result {
+                let $t2
+                // $t2 := (u256)($t0, $t1)
+                /// @src 2:225:246
+                $t2 := $CastU256(x, y)
+                // return $t2
+                /// @src 2:225:246
+                $result := $t2
+            }
+
+            function $Abort(code) {
+                revert(0, 0) // TODO: convention to store code
+            }
+            function $AbortBuiltin() {
+                $Abort(sub(0, 1))
+            }
+            function $AddU256(x, y) -> r {
+                if lt(sub(0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff, x), y) { $AbortBuiltin() }
+                r := add(x, y)
+            }
+            function $MulU256(x, y) -> r {
+                if gt(y, div(0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff, x)) { $AbortBuiltin() }
+                r := mul(x, y)
+            }
+            function $Sub(x, y) -> r {
+                if lt(x, y) { $AbortBuiltin() }
+                r := sub(x, y)
+            }
+            function $Div(x, y) -> r {
+                if eq(y, 0) { $AbortBuiltin() }
+                r := div(x, y)
+            }
+            function $Mod(x, y) -> r {
+                if eq(y, 0) { $AbortBuiltin() }
+                r := mod(x, y)
+            }
+            function $Shr(x, y) -> r {
+                r := shr(x, y)
+            }
+            function $ShlU256(x, y) -> r {
+                r := and(shl(x, y), 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff)
+            }
+            function $Gt(x, y) -> r {
+                r := gt(x, y)
+            }
+            function $Lt(x, y) -> r {
+                r := lt(x, y)
+            }
+            function $GtEq(x, y) -> r {
+                r := or(gt(x, y), eq(x, y))
+            }
+            function $LtEq(x, y) -> r {
+                r := or(lt(x, y), eq(x, y))
+            }
+            function $Eq(x, y) -> r {
+                r := eq(x, y)
+            }
+            function $Neq(x, y) -> r {
+                r := not(eq(x, y))
+            }
+            function $CastU256(hi, lo) -> r {
+                if gt(hi, 0xffffffffffffffffffffffffffffffff) { $AbortBuiltin() }
+                if gt(lo, 0xffffffffffffffffffffffffffffffff) { $AbortBuiltin() }
+                r := add(shl(hi, 128), lo)
+            }
+        }
+    }
+}
+
+
+!! Errors compiling Yul
+
+Warning: Yul is still experimental. Please use the output with care.
```

Comment:
Sorry for not noticing previously, but there is a problem with these source annotations.

Reply:
Not sure whether this is the right solution. It could be a regular comment which just ignores the content? Perhaps one can see from the optimized Yul dump.